### PR TITLE
SC : Refactor bridge contract method parameter

### PR DESCRIPTION
**Pull request by KimKyungup**
## Proposed changes

This PR refactors bridge contract method parameter's name and order.
This change helps next PR (request/handle transaction map).

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues
This PR is a follow-up PR of https://github.com/klaytn/klaytn/pull/57.
So please review the last commit.
https://github.com/klaytn/klaytn/commit/198b25c9ed3b1fa910ce1bf6dc5ec7e5997380c9

After #57 is merged, I will rebase it.

### CODE REVIEW

**Comment by KimKyungup** [comment]
This PR is a follow-up PR of https://github.com/klaytn/klaytn/pull/57.
So please review the last commit.
https://github.com/klaytn/klaytn/commit/198b25c9ed3b1fa910ce1bf6dc5ec7e5997380c9

After #57 is merged, I will rebase it.

File: contracts/bridge/Bridge.sol [hunk 1]
```diff
@@ -46,37 +46,42 @@ contract Bridge is IERC20BridgeReceiver, IERC721BridgeReceiver, Ownable, BridgeF
 
     /**
      * Event to log the withdrawal of a token from the Bridge.
```
Comment (jimni1222) [inline contracts/bridge/Bridge.sol]: Um... I think RequestValueTransfer is for depositing. 
If my thinking is correct, how about edit comment here 'withdrawal' and L49?

File: contracts/bridge/Bridge.sol [hunk 2]
```diff
@@ -46,37 +46,42 @@ contract Bridge is IERC20BridgeReceiver, IERC721BridgeReceiver, Ownable, BridgeF
...
-     * @param amount is the amount for KLAY/TOKEN and the NFT ID for NFT.
+     * @param to is the receiver of the value.
+     * @param tokenAddress Address of token contract the token belong to.
+     * @param valueOrTokenId is the value of KLAY/ERC20 or token ID of ERC721.
      * @param requestNonce is the order number of the request value transfer.
      * @param uri is uri of ERC721 token.
      */
-    event RequestValueTransfer(TokenKind kind,
+    event RequestValueTransfer(
+        TokenKind kind,
```
Comment (laggu) [inline contracts/bridge/Bridge.sol]: what about using TokenType type instead of kind?

File: contracts/bridge/Bridge.sol [hunk 3]
```diff
@@ -105,69 +110,72 @@ contract Bridge is IERC20BridgeReceiver, IERC721BridgeReceiver, Ownable, BridgeF
...
     // handleERC20Transfer sends the token by the request.
     function handleERC20Transfer(
-        uint256 _amount,
+        address _from,
         address _to,
-        address _contractAddress,
+        address _tokenAddress,
+        uint256 _value,
         uint64 _requestNonce,
         uint64 _requestBlockNumber
```
Comment (jimni1222) [inline contracts/bridge/Bridge.sol]: Can I ask what '_requestBlockNumber' or lastHandledRequestBlockNumber is used for?

Comment (KimKyungup) [reply contracts/bridge/Bridge.sol]: `_requestBlockNumber ` is the block (including request value transfer tx)'s number. :)

`lastHandledRequestBlockNumber` is the nonce of handled request event's nonce.

File: contracts/bridge/Bridge.sol (hunk 2 above)
Comment (KimKyungup) [reply contracts/bridge/Bridge.sol]: Done. Thanks. :)

File: contracts/bridge/Bridge.sol (hunk 3 above)
Comment (KimKyungup) [reply contracts/bridge/Bridge.sol]: I will explain offline if you need. :)

File: contracts/bridge/Bridge.sol [hunk 4]
```diff
@@ -45,38 +45,43 @@ contract Bridge is IERC20BridgeReceiver, IERC721BridgeReceiver, Ownable, BridgeF
     }
 
     /**
-     * Event to log the withdrawal of a token from the Bridge.
-     * @param kind The type of token withdrawn (KLAY/TOKEN/NFT).
+     * Event to log the request value transfer from the Bridge.
+     * @param tokenType The type of token withdrawn (KLAY/ERC20/ERC721).
```
Comment (jimni1222) [inline contracts/bridge/Bridge.sol]: How about change 'withdrawn' to another word?

File: contracts/bridge/Bridge.sol (hunk 3 above)
Comment (jimni1222) [reply contracts/bridge/Bridge.sol]: Thank you for explaining!! :)

**Comment by KimKyungup** [comment]
I reflected your review. :) PTAL, again. @laggu @jimni1222 
cc. @jeonghonah

File: contracts/bridge/Bridge.sol [hunk 5]
```diff
@@ -45,38 +45,43 @@ contract Bridge is IERC20BridgeReceiver, IERC721BridgeReceiver, Ownable, BridgeF
...
 
     /**
-     * Event to log the withdrawal of a token from the Bridge.
-     * @param owner Address of the entity that made the withdrawal.ga
-     * @param kind The type of token withdrawn (KLAY/TOKEN/NFT).
-     * @param contractAddress Address of token contract the token belong to.
-     * @param value For KLAY/TOKEN this is the amount.
+     * Event to log the handle value transfer from the Bridge.
+     * @param tokenType is the type of tokens (KLAY/ERC20/ERC721).
+     * @param from is an address of the account who requested the value transfer.
```
Comment (jeonghonah) [inline contracts/bridge/Bridge.sol]:
```suggestion
     * @param from is an address of the account who requests the value transfer.
```

File: contracts/bridge/Bridge.sol [hunk 6]
```diff
@@ -45,38 +45,43 @@ contract Bridge is IERC20BridgeReceiver, IERC721BridgeReceiver, Ownable, BridgeF
...
      */
-    event RequestValueTransfer(TokenKind kind,
+    event RequestValueTransfer(
+        TokenType tokenType,
         address from,
-        uint256 amount,
-        address contractAddress,
         address to,
+        address tokenAddress,
+        uint256 valueOrTokenId,
```
Comment (jeonghonah) [inline contracts/bridge/Bridge.sol]: How about just a `value`?

File: contracts/bridge/Bridge.sol [hunk 7]
```diff
@@ -105,69 +110,72 @@ contract Bridge is IERC20BridgeReceiver, IERC721BridgeReceiver, Ownable, BridgeF
 
     // handleERC20Transfer sends the token by the request.
     function handleERC20Transfer(
-        uint256 _amount,
+        address _from,
         address _to,
-        address _contractAddress,
+        address _tokenAddress,
+        uint256 _value,
```
Comment (jeonghonah) [inline contracts/bridge/Bridge.sol]: I think value is better but we need to use the term consistently.

File: contracts/bridge_test.go [hunk 8]
```diff
@@ -49,6 +52,15 @@ func WaitMined(tx *types.Transaction, backend bind.DeployBackend, t *testing.T)
 		t.Fatal("Failed to WaitMined.", "err", err, "txHash", tx.Hash().String(), "status", receipt.Status)
 		return err
 	}
+
+	if receipt == nil {
+		return errors.New("not found receipt")
```
Comment (jeonghonah) [inline contracts/bridge_test.go]:
```suggestion
		return errors.New("receipt not found")
```

File: contracts/bridge/Bridge.sol (hunk 6 above)
Comment (KimKyungup) [reply contracts/bridge/Bridge.sol]: In ERC721, it means TokenID. So I wrote like this.

**Comment by KimKyungup** [comment]
After rebasing this with #57 , I will let you know guys. :)

File: contracts/bridge/Bridge.sol (hunk 5 above)
Comment (KimKyungup) [reply contracts/bridge/Bridge.sol]: Ad we talked, I keep this code.

**Comment by KimKyungup** [comment]
@jeonghonah @jimni1222 @laggu I rebased this. :) PTAL.

File: contracts/bridge/Bridge.sol (hunk 6 above)
Comment (nogang) [inline contracts/bridge/Bridge.sol]: 'value' and 'tockenId' are like totally different data, so why use them as one variable?

Comment (KimKyungup) [reply contracts/bridge/Bridge.sol]: It is not totally different data.
In ERC20, it is value. In ERC721, it is token ID. :)

**Pull request merged**